From d9169dea2297a50f3f80a78f076e08e82b5837a4 Mon Sep 17 00:00:00 2001
From: Ioana Tibuleac <ioana.tibuleac@freescale.com>
Date: Wed, 14 May 2014 12:46:45 +0300
Subject: [PATCH 686/987] dpa_offload: fix oNIC non-IPSEC Tx traffic for
 ipsec_offload

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

add dts attribute that prevents the pre-SEC OH port to free buffers
when used with an oNIC node

Signed-off-by: Ioana Tibuleac <ioana.tibuleac@freescale.com>
Change-Id: I0d0c4f67973e2661855ed6d30a1d1bea3311f983
Reviewed-on: http://git.am.freescale.net:8181/12333
Reviewed-by: Marian Cristian Rotariu <marian.rotariu@freescale.com>
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
---
 .../staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts    | 1 +
 .../staging/fsl_dpa_offload/dts/b4860qds-usdpaa-shared-interfaces.dts    | 1 +
 .../staging/fsl_dpa_offload/dts/t4240qds-usdpaa-shared-interfaces.dts    | 1 +
 3 files changed, 3 insertions(+)

diff --git a/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts b/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts
index cccbb89..72bf15a 100644
--- a/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts
+++ b/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts
@@ -148,6 +148,7 @@
 			fsl,qman-frame-queues-rx = <7000 2>;
 			fsl,qman-frame-queues-tx = <7008 1>;
 			fsl,oh-ports = <&oh2 &oh3>;
+			fsl,disable_buff_dealloc;
 			local-mac-address = [00 11 22 33 44 88];
 		};
 
diff --git a/drivers/staging/fsl_dpa_offload/dts/b4860qds-usdpaa-shared-interfaces.dts b/drivers/staging/fsl_dpa_offload/dts/b4860qds-usdpaa-shared-interfaces.dts
index a792dce..869e011 100644
--- a/drivers/staging/fsl_dpa_offload/dts/b4860qds-usdpaa-shared-interfaces.dts
+++ b/drivers/staging/fsl_dpa_offload/dts/b4860qds-usdpaa-shared-interfaces.dts
@@ -160,6 +160,7 @@
 			fsl,qman-frame-queues-rx = <7000 2>;
 			fsl,qman-frame-queues-tx = <7008 1>;
 			fsl,oh-ports = <&oh2 &oh3>;
+			fsl,disable_buff_dealloc;
 			local-mac-address = [00 11 22 33 44 88];
 		};
 
diff --git a/drivers/staging/fsl_dpa_offload/dts/t4240qds-usdpaa-shared-interfaces.dts b/drivers/staging/fsl_dpa_offload/dts/t4240qds-usdpaa-shared-interfaces.dts
index a1549b4..54dc5ce 100644
--- a/drivers/staging/fsl_dpa_offload/dts/t4240qds-usdpaa-shared-interfaces.dts
+++ b/drivers/staging/fsl_dpa_offload/dts/t4240qds-usdpaa-shared-interfaces.dts
@@ -170,6 +170,7 @@
 			fsl,qman-frame-queues-rx = <7000 2>;
 			fsl,qman-frame-queues-tx = <7008 1>;
 			fsl,oh-ports = <&dpa_fman0_oh2 &dpa_fman0_oh3>;
+			fsl,disable_buff_dealloc;
 			local-mac-address = [00 11 22 33 44 88];
 		};
 
-- 
1.9.1

